core-java-modules/core-java-io-conversions-2/src/test/java/com/baeldung/csv/ReadCSVInArrayUnitTest.java

```diff
@@ -94,4 +98,51 @@ public void givenCSVFile_whenOpencsv_thenContentsAsExpected() throws IOException
                     .toArray());
         }
     }
+
+    @Test
+    public void givenCSVFile_whenUsingFilesReadAllLinesMethod_thenContentsAsExpected() throws IOException {
+        Path path = Paths.get(CSV_FILE);
+        List<List<String>> records = Files.readAllLines(path)
+          .stream()
+          .map(line -> Arrays.asList(line.split(COMMA_DELIMITER)))
+          .collect(Collectors.toList());
+
+        for (int i = 0; i < EXPECTED_ARRAY.size(); i++) {
+            Assert.assertArrayEquals(EXPECTED_ARRAY.get(i)
+                .toArray(),
+                records.get(i)
+                    .toArray());
+        }
+    }
+
+    @Test
+    public void givenCSVFile_whenUsingFilesNewBufferedReaderMethod_thenContentsAsExpected() throws IOException {
+        Path path = Paths.get(CSV_FILE);
+        List<List<String>> records = Files.newBufferedReader(path)
+          .lines()
+          .map(line -> Arrays.asList(line.split(COMMA_DELIMITER)))
+          .collect(Collectors.toList());
+
+        for (int i = 0; i < EXPECTED_ARRAY.size(); i++) {
+            Assert.assertArrayEquals(EXPECTED_ARRAY.get(i)
+                .toArray(),
+                records.get(i)
+                    .toArray());
+        }
+    }
+
+    @Test
+    public void givenCSVFile_whenUsingFilesLinesMethod_thenContentsAsExpected() throws IOException {
+        Path path = Paths.get(CSV_FILE);
+        List<List<String>> records = Files.lines(path)
+          .map(line -> Arrays.asList(line.split(COMMA_DELIMITER)))
+          .collect(Collectors.toList());
+
+        for (int i = 0; i < EXPECTED_ARRAY.size(); i++) {
+            Assert.assertArrayEquals(EXPECTED_ARRAY.get(i)
+                .toArray(),
+                records.get(i)
+                    .toArray());
+        }
```

Comment:
Here we can use try-with-resources:
```suggestion
        try (Stream<String> lines = Files.lines(Paths.get(CSV_FILE))) {
            List<List<String>> records = lines.map(line -> Arrays.asList(line.split(COMMA_DELIMITER)))
                .collect(Collectors.toList());
            for (int i = 0; i < EXPECTED_ARRAY.size(); i++) {
                Assert.assertArrayEquals(EXPECTED_ARRAY.get(i)
                    .toArray(), records.get(i)
                    .toArray());
            }
        }
```

Reply:
Done